try-greg.py

```diff
@@ -0,0 +1,59 @@
+import sys
+import math
+
+def make_blanket( blanket_radius, blanket_width, blanket_height, shell_thickness ):        
+    cubit.cmd( f"create brick x {blanket_height} y {blanket_width} z {blanket_width}" )
+    shell_volume_id = cubit.get_last_id( "volume" )
+    cubit.cmd( f"modify curve {get_closest_curve_in_volume_to_location( shell_volume_id, (blanket_height/2, blanket_width/2, 0) )} blend radius {blanket_radius}" )
+    cubit.cmd( f"modify curve {get_closest_curve_in_volume_to_location( shell_volume_id, (-blanket_height/2, blanket_width/2, 0) )} blend radius {blanket_radius}" )
+
+    cubit.cmd( f"create sheet offset from surface in vol {shell_volume_id} offset {-1*shell_thickness} " )
+    inner_volume_id = cubit.get_last_id( "volume" )
+
+    before_vol_ids = cubit.get_entities( "volume" )
+    cubit.cmd( f"subtract volume {inner_volume_id} from volume {shell_volume_id} keep_tool" )
+    after_vol_ids = cubit.get_entities( "volume" )
+    inner_volume_id, shell_volume_id = get_operation_return_ids( before_vol_ids, after_vol_ids )
+    
+    cubit.cmd( f"block 1 volume {cubit.get_id_string( shell_volume_id )}" )
+    cubit.cmd( f"block 2 volume {cubit.get_id_string( inner_volume_id )}" )
+    cubit.cmd( "block 1 name 'OuterShell'" )
+    cubit.cmd( "block 2 name 'InnerVolume'" )
+
+def get_closest_curve_in_volume_to_location( vol_id, location ):
+    C = cubit.parse_cubit_list( "curve", f"in volume {vol_id}" )
+    min_dist = float( "inf" )
+    for cid in C:
+        curve = cubit.curve(cid)
+        curve_close_point = curve.closest_point_trimmed( location )
```

Comment:
Again, here we're accessing the `Curve` object, which is a C++ object (see listing below), but we're only using querying functionality - not creating / modifying.

```python
%>C = cubit.curve(1)
%>C
<cubit3.Curve; proxy of <Swig Object of type 'CubitInterface::Curve *' at 0x000001E2A839DD10> >
```